pilot/pkg/bootstrap/server.go

```diff
@@ -1348,3 +1354,16 @@ func (s *Server) initReadinessProbes() {
 		s.addReadinessProbe(name, probe)
 	}
 }
+
+func (s *Server) StreamServerOverloadInterceptor(srv interface{}, ss grpc.ServerStream, info *grpc.StreamServerInfo, handler grpc.StreamHandler) error {
+	current := s.connectionCounter.Add(1)
+	connectionTotal.RecordInt(current)
+	defer func() {
+		connectionTotal.RecordInt(s.connectionCounter.Add(-1))
+	}()
+	if features.ConnectionLimit > 0 && int64(features.ConnectionLimit) < current {
+		return grpcstatus.Errorf(codes.ResourceExhausted, "connection limit exceeded")
```

Comment:
This is incredibly unsafe. Unless you have a fixed number of istiod pods _and clients_, having hard limits around connections without having autoscaling tied to this limit is a recipe to trigger an outage where all connections are denied. It also makes a DOS attack trivial

Reply:
+1. Curious how much memory you have allocated for Istiod and how many total clients you have + and how many clients connected to single Istiod caused Istiod to get OOMKilled?

Reply:
In our test cluster, each instance is limited to 16c40g.（replicas 10+） Each instance usually maintains more than 700 connections, and the memory usage is 22g-26g. When the cluster is thrashing (some failures), it is easy for a single instance to reach the memory limit.

So I think: when the resources of an instance are fixed, the upper limit of clients it can serve is also fixed.